server/webapp/WEB-INF/rails/webpack/models/pipeline_configs/pipeline_config.ts

```diff
@@ -27,6 +27,7 @@ import {Stage} from "./stage";
 export class PipelineConfig extends ValidatableMixin {
   group: Stream<string> = stream("defaultGroup");
   name: Stream<string>;
+  template: Stream<any> = stream();
```

Comment:
Why not `Stream<string>`? I believe the `template` value is just that?

Reply:
@marques-work `template()` can have one of two types of values: `undefined` or `string`

Basically, if template's value is a string, then stages must be empty. If stages is not empty, then template must be undefined. The reason being is that the server returns a validation error if `template` is in the json with any amount of `stages`

Essentially, stages and templates are mutually exclusive

Reply:
`Stream<string>` can hold both `undefined` and `string` :).

Reply:
It's admittedly a minor quibble, but it doesn't hurt to take advantage of typescript's typechecking.